lib/msf/core/post/file.rb

```diff
@@ -470,6 +476,8 @@ def rm_f(*remote_files)
     remote_files.each do |remote|
       if session.type == "meterpreter"
```

Comment:
The Powershell one will remove an empty directory where as the shell one for Windows and Linux will not and the Windows Meterpreter one will throw an error (because `exists?` will evaluate to true for a directory).

Could you please update the Meterpreter and Powershell ones to use `if file?(remote)` (replace the `if exist?(remote)` part for the Meterpreter one. That will cause a directory to be left alone and maintain consistency with the shell ones.

Reply:
okay sure :+1: